lib/Doctrine/DBAL/Connection.php

```diff
@@ -1673,7 +1674,7 @@ public function ping()
             $this->query($this->getDatabasePlatform()->getDummySelectSQL());
 
             return true;
-        } catch (DBALException $e) {
+        } catch (PDOException | DBALException $e) {
```

Comment:
When does a `PDOException` occur? I believe we upcast those, so this is fixing a symptom

Reply:
we may stay with `DBALException` only for case when we will catch `Throwable` and convert them to the DBALException in the `query` method.

https://github.com/doctrine/dbal/blob/master/lib/Doctrine/DBAL/Connection.php#L1011-L1015